v2/api/network/customizations/private_endpoints_extensions.go

```diff
@@ -60,3 +67,99 @@ func (extension *PrivateEndpointExtension) PostReconcileCheck(
 
 	return extensions.PostReconcileCheckResultSuccess(), nil
 }
+
+func (extension *PrivateEndpointExtension) ExportKubernetesResources(
+	ctx context.Context,
+	obj genruntime.MetaObject,
+	armClient *genericarmclient.GenericClient,
+	log logr.Logger,
+) ([]client.Object, error) {
+	// This has to be the current hub storage version. It will need to be updated
+	// if the hub storage version changes.
+	endpoint, ok := obj.(*network.PrivateEndpoint)
+	if !ok {
+		return nil, errors.Errorf("cannot run on unknown resource type %T, expected *network.PrivateEndpoint", obj)
+	}
+
+	// Type assert that we are the hub type. This will fail to compile if
+	// the hub type has been changed but this extension has not
+	var _ conversion.Hub = endpoint
+
+	hasIpConfiguration := configMapSpecified(endpoint)
+	if !hasIpConfiguration {
+		log.V(Debug).Info("no configmap retrieval to perform as operatorSpec is empty")
+		return nil, nil
+	}
+
+	if endpoint.Status.NetworkInterfaces == nil || len(endpoint.Status.NetworkInterfaces) == 0 {
+		log.V(Debug).Info("no configmap retrieval to perform as there is no NetworkInterfaces attached")
+		return nil, nil
+	}
+
+	nicId, err := arm.ParseResourceID(*endpoint.Status.NetworkInterfaces[0].Id)
```

Comment:
No guarantee that this ptr-dereference is safe, I think you need to guard it and early return to avoid a panic?